rest/src/test/kotlin/systems/ajax/englishstudytelegrambot/NatsControllerTest.kt

```diff
@@ -52,189 +49,272 @@ class NatsControllerTest {
-    }
-
-    private fun addLibrariesAndUsers(): List<Library> =
-        listOf(
-            libraryRepository.saveNewLibrary("testLibraryName1", "testTelegramUserId1"),
-            libraryRepository.saveNewLibrary("testLibraryName2", "testTelegramUserId2"),
-            libraryRepository.saveNewLibrary("testLibraryName3", "testTelegramUserId1")
-        )
+    private lateinit var adminService: AdminService
 
     private fun createEmptyAdditionalInfoAboutWord() = AdditionalInfoAboutWord("", "", "", "")
 
     @Test
     fun testGetAllUserNatsController() {
-
-        val libraries = addLibrariesAndUsers()
-
-        val message: Message = doRequest(GET_ALL_USERS_SUBJECT, GetAllUsersRequest.getDefaultInstance().toByteArray())
-
+        // GIVEN
+        val nanoTime = System.nanoTime().toString()
+        val libraryName1 = "$nanoTime libraryName1"
+        val libraryName2 = "$nanoTime libraryName2"
+        val telegramUserId1 = "$nanoTime userIdName1"
+        val telegramUserId2 = "$nanoTime userIdName2"
+
+        val library1 = libraryRepository.saveNewLibrary(libraryName1, telegramUserId1).block()
+        val library2 = libraryRepository.saveNewLibrary(libraryName2, telegramUserId2).block()
+
+        // WHEN
+        val message = doRequest(GET_ALL_USERS_SUBJECT, GetAllUsersRequest.getDefaultInstance().toByteArray())
         val telegramUserIdsList: List<String> =
             GetAllUsersResponse.parser().parseFrom(message.data).success.telegramUserIdsList
 
-        //testTelegramUserId1, testTelegramUserId2
-        Assertions.assertIterableEquals(
-            adminService.getAllUsers().map(UserDtoResponse::telegramUserId), telegramUserIdsList
-        )
+        //THEN
+        StepVerifier.create(adminService.getAllUsers().map { it.telegramUserId }.filter { it.contains(nanoTime) })
+            .expectNext(telegramUserId1)
+            .expectNext(telegramUserId2)
+            .verifyComplete()
     }
 
     @Test
     fun testGetAllLibrariesNatsController() {
-        val libraries = addLibrariesAndUsers()
+        // GIVEN
+        val nanoTime = System.nanoTime().toString()
+        val libraryName1 = "$nanoTime libraryName1"
+        val libraryName2 = "$nanoTime libraryName2"
+        val telegramUserId = "$nanoTime userIdName"
+
+        val library1 = libraryRepository.saveNewLibrary(libraryName1, telegramUserId).block()
+        val library2 = libraryRepository.saveNewLibrary(libraryName2, telegramUserId).block()
 
+        // WHEN
         val message = doRequest(GET_ALL_LIBRARIES_SUBJECT, GetAllLibrariesRequest.getDefaultInstance().toByteArray())
         val libraryList =
             GetAllLibrariesResponse.parser().parseFrom(
                 message.data
             ).success.librariesList
 
-        Assertions.assertIterableEquals(libraries.map(Library::toLibraryResponse), libraryList)
+        //THEN
+        Assertions.assertTrue(libraryList.filter { it.name.contains(nanoTime) }.size == 2)
```

Comment:
Care naming! libraryList should be something like expectedLibraryList
Remember that mistake in test may cause problems. And if one time you have 2 similar names and you confuse them - you get problem that not easy to find. So keep one variable as expected*** every time